Library/ViewModels/CommentRepliesViewModelTests.swift

```diff
@@ -172,6 +172,93 @@ internal final class CommentRepliesViewModelTests: TestCase {
+      loggedInUser: User.template,
+      segmentClient: segmentClient,
+      appTrackingTransparency: MockAppTrackingTransparency()
+    )
+
+    withEnvironment(currentUser: .template, ksrAnalytics: ksrAnalytics) {
+      self.vm.inputs
+        .configureWith(
+          comment: .template,
+          project: .template,
+          update: nil,
+          inputAreaBecomeFirstResponder: false,
+          replyId: nil
+        )
+
+      self.vm.inputs.viewDidLoad()
+      self.vm.inputs.viewDidAppear()
+
+      self.vm.inputs.viewDidLoad()
+
+      self.vm.inputs.blockUser(id: "111")
+
+      self.scheduler.advance()
+
+      XCTAssertEqual(segmentClient.events, ["CTA Clicked"])
+
+      XCTAssertEqual(segmentClient.properties(forKey: "session_user_is_logged_in", as: Bool.self), [true])
+      XCTAssertEqual(segmentClient.properties(forKey: "user_uid", as: String.self), ["\(User.template.id)"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_cta"), ["block_user"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_page"), ["project"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_section"), ["comments"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_type"), ["initiate"])
+      XCTAssertEqual(segmentClient.properties(forKey: "interaction_target_uid"), ["111"])
+    }
+  }
+
+  func testTrackUserBlockedFromComment_ConfirmEventEmitted() {
+    let segmentClient = MockTrackingClient()
+    let ksrAnalytics = KSRAnalytics(
+      config: .template,
+      loggedInUser: User.template,
+      segmentClient: segmentClient,
+      appTrackingTransparency: MockAppTrackingTransparency()
+    )
+
+    withEnvironment(
+      apiService: MockService(blockUserResult: .success(EmptyResponseEnvelope())),
+      currentUser: .template,
+      ksrAnalytics: ksrAnalytics
+    ) {
+      self.vm.inputs
+        .configureWith(
+          comment: .template,
+          project: .template,
+          update: nil,
+          inputAreaBecomeFirstResponder: false,
+          replyId: nil
+        )
+
+      self.vm.inputs.viewDidLoad()
+      self.vm.inputs.viewDidAppear()
+
+      self.vm.inputs.viewDidLoad()
+
+      self.vm.inputs.blockUser(id: "111")
+
+      self.scheduler.advance()
+
+      XCTAssertEqual(segmentClient.events, ["CTA Clicked", "CTA Clicked"])
+
+      XCTAssertEqual(
+        segmentClient.properties(forKey: "session_user_is_logged_in", as: Bool.self),
+        [true, true]
+      )
+      XCTAssertEqual(segmentClient.properties(forKey: "user_uid", as: String.self), ["1", "1"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_cta"), ["block_user", "block_user"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_page"), ["project", "project"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_section"), ["comments", "comments"])
+      XCTAssertEqual(segmentClient.properties(forKey: "context_type"), ["initiate", "confirm"])
```

Comment:
Since this is checking for both an `initiate` and `confirm` event, do we need the previous test? It feels redundant to me, so unless it's doing something I haven't noticed, I think we should delete it (and rename this test to `testTrackUserBlockedFromComment_EventsEmitted`).

This applies to the other tests as well.